api/server/router/container/backend.go

```diff
@@ -43,7 +43,7 @@ type stateBackend interface {
 	ContainerStop(name string, seconds int) error
 	ContainerUnpause(name string) error
 	ContainerUpdate(name string, hostConfig *container.HostConfig) ([]string, error)
-	ContainerWait(name string, timeout time.Duration) (int, error)
+	ContainerWait(name string, timeout time.Duration, waitForRemoval bool) (int, error)
```

Comment:
@WeiZhang555 why?

If there's a tty and `--rm` wasn't specified (but a restart policy was) we would still get the "die" event and the client can exit then. No need for a timeout. Or am I missing a case?

If we want to change the behavior where detaching from a container started with `run -ti` doesn't return, it should probably be a different PR (and can be solved by adding a "detach" event I guess).